### PR TITLE
options to open runme as text editor and text-edit md as runme

### PR DESCRIPTION
**Pull request by lorenzejay**
addresses https://github.com/stateful/vscode-runme/issues/51
openwith-editor-menu-v2

Screen Shot 2022-11-15 at 8 17 41 AM

Screen Shot 2022-11-15 at 8 17 48 AM

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -70,6 +71,26 @@
...
+        "title": "Open markdown as runme",
+        "icon": {
+          "light": "assets/runme-logo-light.svg",
+          "dark": "assets/runme-logo-dark.svg"
+        }
+      },
+      {
+        "command": "runme.openSplitViewAsMarkdownText",
+        "category": "Runme",
+        "title": "Open runme notebook as text editor",
```
Comment (christian-bromann) [inline package.json:92]: (non native english speaker here 👋 ) I am not sure but doesn't make this more sense
```suggestion
        "title": "Open Runme notebook in text editor",
```

File: package.json [hunk 2]
```diff
@@ -70,6 +71,26 @@
...
+      },
+      {
+        "command": "runme.resetEnv",
+        "category": "Runme",
+        "title": "Reset Environment Variables for current VS Code session"
+      },
+      {
+        "command": "runme.openAsRunmeNotebook",
+        "category": "Runme",
+        "title": "Open markdown as runme",
```
Comment (christian-bromann) [inline package.json:83]: Also should we avoid showing this command in the command palette? I assume this command only makes sense in certain context so we should ensure it is only shown there.

```suggestion
        "title": "Open markdown as Runme",
```

File: src/extension/extension.ts [hunk 3]
```diff
@@ -9,7 +9,21 @@ import { PidStatusProvider } from './provider/pid'
...
+  openSplitViewAsMarkdownText 
+} from './commands'
+
+export function activate(context: ExtensionContext) {
+  context.subscriptions.push(
+    commands.registerCommand('runme.openSplitViewAsMarkdownText', openSplitViewAsMarkdownText),
+    commands.registerCommand('runme.openAsRunmeNotebook', openAsRunmeNotebook),
+  )
+}
+export function deactivate() {}
```
Comment (christian-bromann) [inline src/extension/extension.ts:26]: Activation happens in `src/extension/index.ts`

```suggestion
```

File: src/extension/commands/index.ts [hunk 4]
```diff
@@ -32,3 +32,14 @@ export async function runCLICommand (cell: NotebookCell) {
...
+  vscode.window.showNotebookDocument(doc, {
+    viewColumn: vscode.ViewColumn.Beside
+  })
+}
+
+export function openSplitViewAsMarkdownText (doc: TextDocument) {
+  vscode.window.showTextDocument(doc, {
+    viewColumn: vscode.ViewColumn.Beside
+  })
+}
```
Comment (christian-bromann) [inline src/extension/commands/index.ts:45]: Can we have tests for these?

File: src/extension/commands/index.ts [hunk 5]
```diff
@@ -33,13 +33,13 @@ export async function runCLICommand (cell: NotebookCell) {
...
+  window.showNotebookDocument(doc, {
+    viewColumn: 2
   })
 }
 
 export function openSplitViewAsMarkdownText (doc: TextDocument) {
-  vscode.window.showTextDocument(doc, {
-    viewColumn: vscode.ViewColumn.Beside
+  window.showTextDocument(doc, {
+    viewColumn: 2
```
Comment (christian-bromann) [inline src/extension/commands/index.ts:43]: Can we keep `ViewColumn.Beside` and import it above?

File: tests/extension/commands/index.test.ts [hunk 6]
```diff
@@ -60,3 +67,13 @@ test('runCLICommand if CLI is installed', async () => {
   expect(terminal.show).toBeCalledTimes(1)
   expect(terminal.sendText).toBeCalledWith('runme run foobar --chdir="/foo"')
 })
+
+test('open markdown as runme notebook', (file: NotebookDocument) => {
+  openAsRunmeNotebook(file)
+  expect(window.showNotebookDocument).toBeCalledWith(file, { viewColumn: 2})
```
Comment (christian-bromann) [inline tests/extension/commands/index.test.ts:73]: To assert just extend the mock with arbitrary values.

Comment (lorenzejay) [reply tests/extension/commands/index.test.ts:73]: Do I extend in the __mocks__ or direct from the expect with extend? ex) `expect.extend...`

I am getting this as my error
Screen Shot 2022-11-15 at 2 45 26 PM

Comment (lorenzejay) [reply tests/extension/commands/index.test.ts:73]: this doesnt seem to work
Screen Shot 2022-11-15 at 2 45 26 PM

Comment (christian-bromann) [reply tests/extension/commands/index.test.ts:73]: You have to mock `ViewColumn.Beside` which means to create and export an enum in https://github.com/stateful/vscode-runme/blob/main/__mocks__/vscode.ts, then in your test do:

```ts
import { ViewColumn } from 'vscode'
...
expect(window.showNotebookDocument).toBeCalledWith(file, { viewColumn: ViewColumn.Beside })
```

**Comment by lorenzejay** [comment]
@christian-bromann thank you for the suggestions, PR resolved now.

File: package.json [hunk 7]
```diff
@@ -70,13 +71,49 @@
...
     "menus": {
       "commandPalette": [
         {
           "command": "runme.openTerminal",
           "when": "false"
+        }, 
+        {
+          "command": "runme.openAsRunmeNotebook",
+          "when": "false"
+        }
```
Comment (christian-bromann) [inline package.json:105]: Should both commands be added to the command palette?

```suggestion
        }, 
        {
          "command": "runme.openAsRunmeNotebook",
          "when": "false"
        }, 
        {
          "command": "runme.openSplitViewAsMarkdownText",
          "when": "false"
        }
```

**Review by christian-bromann — APPROVED** [review]
Last comment, feel free to ignore

LGTM 👍

**Comment by lorenzejay** [comment]
@christian-bromann will merge after this test passes. Removed openSplitViewAsMarkdownText from command pallete on the latest commit

**Pull request merged**